packages/test/test-service-load/src/gcDataStores.ts

```diff
@@ -794,45 +794,80 @@ export class MultiCollabDataObject extends SingleCollabDataObject implements IGC
 		const partnerRunId2 = ((myRunId + halfClients + 1) % config.testConfig.numClients) + 1;
 		const partnerId1 = `client${partnerRunId1}`;
 		const partnerId2 = `client${partnerRunId2}`;
+
 		/**
-		 * Set up an event handler that listens for changes in the data object map meaning that a child data object
-		 * was referenced or unreferenced by a client.
+		 * Set up an event listener that will run / stop activity based on the activities of the partner.
+		 * If a partner referenced a data store or attachment blob, run activity on the corresponding local object.
+		 * If a partner unreferenced a data store or attachment blob, stop activity on the corresponding local object.
 		 */
-		const partnerActivityRunner = async (changed: IValueChanged, local: boolean) => {
+		const partnerActivityRunner = async (
+			changed: IValueChanged,
+			local: boolean,
+			activityObjectMap: SharedMap,
+			partnerIds: string[],
+		) => {
 			if (local) {
 				return;
 			}
 
-			// Only collaborate with two other partner clients. If we collaborate with all clients, there would be too
-			// many ops and we might get throttled.
-			if (!changed.key.startsWith(partnerId1) && !changed.key.startsWith(partnerId2)) {
+			// Collaborate with the partners clients specified in partnerIds.
+			let collaborate: boolean = false;
+			for (const partnerId of partnerIds) {
+				if (changed.key.startsWith(partnerId)) {
+					collaborate = true;
+					break;
+				}
+			}
+			if (!collaborate) {
 				return;
 			}
 
-			// If a new data object was referenced, run our corresponding local data object.
-			// If a data object was unreferenced, stop running our corresponding local data object.
-			if (this.dataObjectMap.has(changed.key)) {
-				const dataObjectHandle = this.dataObjectMap.get<IFluidHandle<IGCActivityObject>>(
+			// If a new object was referenced, run our corresponding local data object.
+			// If an object was unreferenced, stop running our corresponding local data object.
+			if (activityObjectMap.has(changed.key)) {
+				const activityObjectHandle = activityObjectMap.get<IFluidHandle<IGCActivityObject>>(
 					changed.key,
 				);
-				assert(dataObjectHandle !== undefined, `Could not find handle for ${changed.key}`);
-				const dataObject = await dataObjectHandle.get();
-				const result = await dataObject.run(
+				assert(
+					activityObjectHandle !== undefined,
+					`Could not find handle for ${changed.key}`,
+				);
+				const activityObject = await activityObjectHandle.get();
+				const result = await activityObject.run(
```

Comment:
I thought `run` returned whether it was `done` running or something.  This looks like it's returning `success`?

Reply:
Well, if it compeleted it returns true. If it returns before it completed, it returns false. For example, a data store is supposed to send 1000 ops but if it returns before that due to some reason, it will return false.